### DISCUSSION TITLE
Extract parts of a PR into smaller PRs

### DISCUSSION THREAD

**Discussion by GromNaN** [community/community#102038] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

Sometimes, when you start working on a subject, you find other things that can be quickly fixed. So it's easy to do the change directly in your working branch. But you end with a bulk of unrelated change and the PR is bloated.

What I would like, it to be able to extract some changes from a PR, to create a new PR. With or without reverting this changes from the original PR.

Using git cli, it's possible to create a new branch and checkout each file at the state they are in the other branch. But that's not very convenient.

**Comment by chavachai** [comment]
I am not entirely sure, but this can be one approach to try. Maybe you can first create a new branch for specific changes using 'git checkout' and then 'cherry-pick' the commits you want to include in your new pull request. Once resolved conflicts, then commit the changes and push the new branch. Finally, create a new pull request! It allows us to separate specific changes from the original pull request and create a more focused pull request without affecting the existing one.

**Reply by GromNaN** [reply]
That would work if changes have been made with granular commits at first. But we know that when digging, this is not always possible. Especially if the need to split the PR comes during the review.

**Comment by yarikoptic** [comment]
Indeed it is like filing a new issue from a comment from another issue/PR (feature now present) -- here while going through review of a PR I would have loved if I could similar to `suggestion` introduce an annotation to take collection of hunks and introduce them as a separate PR. Common use case is where feature-specific PR includes some generic changes worth a separate one.

Options could be
- take entire commit with owners etc if hunk covers entire commit
- squash information from all the commits with authors etc in comments similarly to squashing a branch etc while allowing to edit it before actually preparing that other PR
